internal/pkg/userdata/userdata.go

```diff
@@ -169,27 +163,22 @@ func (kdm *Kubeadm) MarshalYAML() (interface{}, error) {
 		kdm.bootstrap = false
 	}
 
-	aux.Configuration = string(b)
+	kdm.ConfigurationStr = string(b)
 
-	return aux, nil
+	type KubeadmAlias Kubeadm
```

Comment:
this is Go trick - types are almost equal, but any methods defined on the type are gone (including MarshalYAML, so that we don't fall into endless recursion)

Reply:
Is it worth doing `type KubeadmAlias = Kubeadm`?

Reply:
Meh probably not. It is a temporary struct that we don't actually use anywhere else.

Reply:
I belive if we do `KubeadmAlias = Kubeadm`, this is going to blow up, as methods would be available and it's going to be endless recursion as type equality is stronger than alias